plugin/caffe/caffe_data_iter.cc

```diff
@@ -0,0 +1,217 @@
+/*!
+ * Copyright (c) 2015 by Contributors
+ * \file caffe_data_iter.cc
+ * \brief register mnist iterator
+*/
+#include <caffe/proto/caffe.pb.h>
+#include <dmlc/logging.h>
+#include <dmlc/parameter.h>
+#include <mxnet/operator.h>
+#include <atomic>
+
+#include "../../src/operator/operator_common.h"
+#include "caffe_common.h"
+#include "caffe_stream.h"
+#include "caffe_fieldentry.h"
+#include "caffe_blob.h"
+#include "../../src/io/inst_vector.h"
+#include "../../src/io/iter_prefetcher.h"
+#include "../../src/operator/cast-inl.h"
+
+namespace mxnet {
+namespace io {
+
+struct CaffeDataParam : public dmlc::Parameter<CaffeDataParam> {
+  /*! \brief protobuf text */
+  ::caffe::LayerParameter prototxt;
+  /*! \brief maximum iteration count */
+  int max_iterations;
```

Comment:
Better name for which one? Max_iterations? This item is different from epoch. For basic MXMnistIter (or whatever it's called), the iterator class knows the number of images/blobs in the data set, and simply stops (during each epoch) when it processes all.of the images once. This isn't the case with caffe -- caffe data layer doesn't seem to have any mechanism to either designate or discover the number of images/blobs and simply iterates as many times as requested (or maybe based on other parameters such as accuracy reached -- i didn't analyze beyond max other in caffe).  Epoch appears to be set in the Python layer.  Please advise.